### PR TITLE
[TE] Refactor classifier interface to simplify implementation

### PR DESCRIPTION
**Pull request by cyenjung**
The refactoring has two purpose:
1. Users do not need to update anomaly result manually: The classify method now returns only the issue type for the given anomaly. The update is performed by classification pipeline.
2. Simplify the implementation of classify method because it takes one main anomaly at at time. Therefore, users do not need to manually go through the list of main anomaly.

Test on local controllers.

### CODE REVIEW

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/classification/ClassificationTaskRunner.java [hunk 1]
```diff
@@ -195,22 +198,80 @@ private void runTask(long windowStart, long windowEnd) {
...
+    }
+    String originalIssueType = null;
+    if (anomalyProperties.containsKey(ISSUE_TYPE_KEY)) {
+      originalIssueType = anomalyProperties.get(ISSUE_TYPE_KEY);
+    }
+    if (!Objects.equals(issueType, originalIssueType)) {
+      anomalyProperties.put(ISSUE_TYPE_KEY, issueType);
+      updated = true;
+    }
+    return updated;
```
Comment (apucher) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/classification/ClassificationTaskRunner.java]: just return true/false and get rid of boolean updated

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/classification/ClassificationTaskRunner.java [hunk 2]
```diff
@@ -195,22 +198,80 @@ private void runTask(long windowStart, long windowEnd) {
...
+            existingOverlappedAuxAnomalies.addAll(overlappedAuxAnomalies);
+          } else {
+            auxiliaryAnomalies.put(metricName, overlappedAuxAnomalies);
+          }
+        }
+        // Get and update issue type for the current main anomalies
+        String issueType = anomalyClassifier.classify(mainAnomalyResult, auxiliaryAnomalies);
+        boolean updated = updateIssueTypeForAnomalyResult(mainAnomalyResult, issueType);
+        if (updated) {
+          updatedMainAnomaliesByDimension.add(mainAnomalyResult);
```
Comment (apucher) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/classification/ClassificationTaskRunner.java]: just skip defining updated

Comment (cyenjung) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/classification/ClassificationTaskRunner.java]: Fixed.

File: thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/classification/ClassificationTaskRunner.java (hunk 1 above)
Comment (cyenjung) [inline thirdeye/thirdeye-pinot/src/main/java/com/linkedin/thirdeye/anomaly/classification/ClassificationTaskRunner.java]: Done!

**Pull request merged**